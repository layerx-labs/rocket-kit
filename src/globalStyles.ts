import { createGlobalStyle } from 'styled-components/macro';
import { rem} from "polished";
import { typography, colors } from './ions/variables';
import { device } from './ions/breakpoints';

const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
    scroll-behavior: smooth;
  }

  body {
    margin: 0;
    font-family: 'Space Grotesk', Verdana, Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: ${typography.regular};
    color: ${colors.black};
    font-feature-settings: 'ss04' on;
    -webkit-font-smoothing: antialiased;
  }

  h1,
  h1 span,
  h2,
  h2 span {
    margin: 0;
    font-weight: ${typography.regular};
  }

  h1,
  h1 span {
    font-family: 'Laforgestencil', Verdana, Arial, Helvetica, sans-serif;
    font-size: 3.5rem;
    line-height: 1.15;
  }

  h2,
  h2 span {
    font-size: 2rem;
    line-height: 1.2;
  }

  h3,
  h4 {
    margin: 0;
    font-size: 1rem;
    font-weight: 400;
  }

  span {
    font-family: ${typography.defaultFont};
    font-size: ${rem(typography.defaultSize)};
    font-feature-settings: 'ss04' on;
  }

  p {
    margin: 0;
    font-size: 1rem;
    line-height: 1.3;

    &:not(:last-child) {
      margin-bottom: ${rem('10px')};
    }
  }

  pre {
    border: 1px solid ${colors.grey500};
    border-radius: 4px;
    background-color: ${colors.grey300};
    padding: ${rem('15px')};
  }

  @media ${device.l} {
    h1,
    h1 span {
      font-size: 6rem;
      line-height: 1.1;
    }

    h2,
    h2 span {
      font-size: 2.5rem;
    }
  }
`;

export default GlobalStyle;
