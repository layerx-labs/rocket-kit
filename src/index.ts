// Atoms
export { default as Avatar } from './atoms/avatar';
export { default as Button } from './atoms/button';
export { default as ButtonLink } from './atoms/button-link';
export { default as Checkbox } from './atoms/checkbox';
export { default as Icon } from './atoms/icon';
export { default as Spinner } from './atoms/spinner';
export { default as Tag } from './atoms/tag';

// Molecules
export { default as NoteCard } from './molecules/note-card';
