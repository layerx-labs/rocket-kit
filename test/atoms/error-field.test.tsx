import React from 'react';
import { render, screen } from '@testing-library/react';
import { ErrorField } from '../../src';

describe('ErrorField', () => {
  it('renders', () => {
    const { asFragment } = render(<ErrorField error={'operation failed.'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('ErrorField error has correct text', async () => {
    const fieldText = 'Failed to save name';
    render(<ErrorField error={fieldText} color="danger" />);
    const foundText = await screen.getByText(fieldText);
    expect(foundText).toBeTruthy();
  });

  it('ErrorField success has correct text', async () => {
    const fieldText = 'Name save!';
    render(<ErrorField error={fieldText} color="success" />);
    const foundText = await screen.getByText(fieldText);
    expect(foundText).toBeTruthy();
  });
});
