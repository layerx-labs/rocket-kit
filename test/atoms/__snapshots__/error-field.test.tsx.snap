// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorField renders 1`] = `
<DocumentFragment>
  <span
    class="sc-fubCfw laRIMs"
    color="danger"
  >
    operation failed.
  </span>
</DocumentFragment>
`;
